lib/fluent/plugin/out_papertrail.rb

```diff
@@ -1,30 +1,84 @@
-require 'syslog_protocol'
-require 'fluent/output'
-
 module Fluent
-  class Papertrail < Fluent::Output
-    attr_accessor :socket, :papertrail_host, :papertrail_port
+  class Papertrail < Fluent::BufferedOutput
+    attr_accessor :socket
+
+    # if left empty in fluent config these config_param's will error
+    config_param :papertrail_host, :string
+    config_param :papertrail_port, :integer
+    # overriding default flush_interval (60 sec) from Fluent::BufferedOutput
+    config_param :flush_interval, :time, default: 1
 
+    # register as 'papertrail' fluent plugin
     Fluent::Plugin.register_output('papertrail', self)
 
+    def initialize
+      super
+      require 'syslog_protocol'
```

Comment:
Why require here?

Reply:
Seemed to be a style in some of the other plugins I looked at, but I see no reason not to just put it at the top of the file. Then we can also ditch the def initialize super end altogether. Is that what you're thinking?

Reply:
* re `require` would love to see a link to those b/c maybe there's a great reason that I'm just missing, or an in-context reason that would help me learn something about this ecosystem.

* Are you saying that the call to `super` isn't necessary? I had assumed there was some kind of setup you needed in the parent class or something.